.github/workflows/e2e-test.yml

```
@@ -85,17 +85,52 @@ jobs:
           set -ex
           REPO_NAME=$(echo "${GITHUB_REPOSITORY}" | cut -d'/' -f2)
           REPO_URL="https://github.com/omec-project/${REPO_NAME}"
+          NAMESPACE="aether-5gc"
+          
           git clone ${REPO_URL} ${REPO_NAME} && cd ${REPO_NAME}
-          PR_NUMBER=$(echo "${GITHUB_REF}" | awk -F'/' '{print $3}')
-          git fetch origin pull/${PR_NUMBER}/merge:pr-${PR_NUMBER}
-          git checkout pr-${PR_NUMBER}
+
+          if [[ "${GITHUB_REF}" == refs/pull/* ]]; then
+            PR_NUMBER=$(echo "${GITHUB_REF}" | awk -F'/' '{print $3}')
+            git fetch origin pull/${PR_NUMBER}/merge:pr-${PR_NUMBER}
+            git checkout pr-${PR_NUMBER}
+          else
+            BRANCH_NAME=$(basename "${GITHUB_REF}")
+            git checkout "${BRANCH_NAME}"
+          fi
+
           if [ "${REPO_NAME}" = "webconsole" ]; then
-            REPO_NAME="webui"
+            IMAGE_NAME="webui"
+          else
+            IMAGE_NAME="${REPO_NAME}"
           fi
-          docker build -t ${REPO_NAME}:testing .
-          kubectl set image deployment/${REPO_NAME} ${REPO_NAME}=${REPO_NAME}:testing -n aether-5gc
-          kubectl -n aether-5gc patch deployment ${REPO_NAME} \
-          -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"${REPO_NAME}\",\"image\":\"${REPO_NAME}:testing\",\"imagePullPolicy\":\"Never\"}]}}}}"
+
+          docker build -t "${IMAGE_NAME}:testing" .
+          
+          # Save and import Docker image into containerd (used by RKE2) because RKE2 does not use Docker runtime.
+          TMP_DIR="${RUNNER_TEMP:-/tmp}"
+          docker save "${IMAGE_NAME}:testing" -o "${TMP_DIR}/${IMAGE_NAME}.tar"
+          sudo ctr --address /run/k3s/containerd/containerd.sock --namespace k8s.io images import "${TMP_DIR}/${IMAGE_NAME}.tar"
+          sudo ctr --address /run/k3s/containerd/containerd.sock --namespace k8s.io images ls | grep "${IMAGE_NAME}:testing"
+
+          # Patch K8s deployment with imagePullPolicy Never
+          kubectl -n "$NAMESPACE" patch deployment "${REPO_NAME}" --type='merge' -p "$(cat <<EOF
+          {
+            "spec": {
+              "template": {
+                "spec": {
+                  "containers": [
+                    {
+                      "name": "${IMAGE_NAME}",
+                      "image": "docker.io/library/${IMAGE_NAME}:testing",
+                      "imagePullPolicy": "Never"
+                    }
+                  ]
+                }
+              }
+            }
+          }
+          EOF
+          )"
```

Should not this be as follows?

```suggestion
          kubectl -n "$NAMESPACE" patch deployment "${REPO_NAME}" --type='merge' -p '$(cat <<EOF
          {
            "spec": {
              "template": {
                "spec": {
                  "containers": [
                    {
                      "name": "${IMAGE_NAME}",
                      "image": "docker.io/library/${IMAGE_NAME}:testing",
                      "imagePullPolicy": "Never"
                    }
                  ]
                }
              }
            }
          }
          EOF
          )'
```